### PR TITLE
Handle some deprecation warnings

### PR DESCRIPTION
**Pull request by nik9000**
Suppress lots of them with comments. Fix a few.

### CODE REVIEW

**Comment by bleskes** [comment]
Lgtm. Thx @nik9000!

File: core/src/main/java/org/elasticsearch/indices/analysis/PreBuiltTokenFilters.java [hunk 1]
```diff
@@ -108,26 +109,24 @@ public TokenStream create(TokenStream tokenStream, Version version) {
...
             }
         }
     },
 
     TRIM(CachingStrategy.LUCENE) {
         @Override
+        @SuppressWarnings("deprecation") // For backwards compatibility
         public TokenStream create(TokenStream tokenStream, Version version) {
             if (version.luceneVersion.onOrAfter(org.apache.lucene.util.Version.LUCENE_4_4_0)) {
                 return new TrimFilter(tokenStream);
```
Comment (rjernst) [inline core/src/main/java/org/elasticsearch/indices/analysis/PreBuiltTokenFilters.java]: Are you planning to backport this to 2.x? At least in master we can just remove the condtion, it should always be TrimFilter?

File: core/src/main/java/org/elasticsearch/indices/analysis/PreBuiltTokenFilters.java [hunk 2]
```diff
@@ -108,26 +109,24 @@ public TokenStream create(TokenStream tokenStream, Version version) {
 
     STOP(CachingStrategy.LUCENE) {
         @Override
+        @SuppressWarnings("deprecation") // For backwards compatibility
         public TokenStream create(TokenStream tokenStream, Version version) {
             if (version.luceneVersion.onOrAfter(org.apache.lucene.util.Version.LUCENE_4_4_0)) {
                 return new StopFilter(tokenStream, StopAnalyzer.ENGLISH_STOP_WORDS_SET);
```
Comment (rjernst) [inline core/src/main/java/org/elasticsearch/indices/analysis/PreBuiltTokenFilters.java]: Same here, always StopFilter on master.

**Comment by rjernst** [comment]
LGTM too.

File: core/src/main/java/org/elasticsearch/indices/analysis/PreBuiltTokenFilters.java (hunk 1 above)
Comment (nik9000) [inline core/src/main/java/org/elasticsearch/indices/analysis/PreBuiltTokenFilters.java]: No I didn't plan to backport it. It looks like I can drop anything in master that checks for Lucene version less than 4.6.1. Sound right to you?

**Comment by nik9000** [comment]
Pushed another commit that just drops support for the pre-1.0 versions rather than suppresses the warnings.

File: core/src/main/java/org/elasticsearch/indices/analysis/PreBuiltTokenFilters.java [hunk 3]
```diff
@@ -210,13 +189,12 @@ public TokenStream create(TokenStream tokenStream, Version version) {
 
     NGRAM(CachingStrategy.LUCENE) {
         @Override
+        @SuppressWarnings("deprecation") // For backwards compatibility
         public TokenStream create(TokenStream tokenStream, Version version) {
             if (version.luceneVersion.onOrAfter(org.apache.lucene.util.Version.LUCENE_4_4_0)) {
                 return new NGramTokenFilter(tokenStream);
```
Comment (rjernst) [inline core/src/main/java/org/elasticsearch/indices/analysis/PreBuiltTokenFilters.java]: This entire if/else should be removed?

File: core/src/main/java/org/elasticsearch/index/store/Store.java [hunk 4]
```diff
@@ -118,6 +118,8 @@
  * </pre>
  */
 public class Store extends AbstractIndexShardComponent implements Closeable, RefCounted {
+    @SuppressWarnings("deprecation")
+    private static final Version FIRST_LUCENE_CHECKSUM_VERSION = Version.LUCENE_4_8_0;
```
Comment (rjernst) [inline core/src/main/java/org/elasticsearch/index/store/Store.java]: Do we need this at all?

**Comment by rjernst** [comment]
The minimum version is Lucene 5 for master (or rather that is the intention in master, with an eventual 6.0 release of lucene). If we are assuming no release of master can happen without lucene 6, then removing any conditions checking before Lucene 5 should be ok.

**Comment by nik9000** [comment]
I was mixed up when I last mentioned versions. In master we have to be able to open up indices from 2.x  which means Lucene 5.2.1 and onwards so anything older than that is fine to remove. Right? So we can remove the old checksum support entirely.

File: core/src/main/java/org/elasticsearch/indices/analysis/PreBuiltTokenFilters.java (hunk 3 above)
Comment (nik9000) [inline core/src/main/java/org/elasticsearch/indices/analysis/PreBuiltTokenFilters.java]: Indeed. Removing.

**Comment by rjernst** [comment]
Yes, although I would not look for the specific minor release, but only the major version (while you are correct we released 2.0 with 5.2.1, I think it is much simpler to think about major versions alone).

**Comment by nik9000** [comment]
Right. I'm working on these deprecation and other Xlint things to "relax" so I'll probably put a bow on this PR and set it aside and take up the hunt for the rest of the stuff in another PR soonish. I'll remove that 4.4 ngram filter and remove my `SuppressWarnings` on `FIRST_LUCENE_CHECKSUM_VERSION` so it'll be real obvious the next time someone goes to hunt deprecations. They (or I) can remove the old checksum support then.

**Comment by nik9000** [comment]
And done. @rjernst can you have another look?

File: core/src/main/java/org/elasticsearch/indices/analysis/PreBuiltTokenizers.java [hunk 5]
```diff
@@ -49,6 +47,7 @@
 
     STANDARD(CachingStrategy.LUCENE) {
         @Override
+        @SuppressWarnings("deprecation") // Backwards compatibility
         protected Tokenizer create(Version version) {
             if (version.luceneVersion.onOrAfter(org.apache.lucene.util.Version.LUCENE_4_7_0)) {
```
Comment (rjernst) [inline core/src/main/java/org/elasticsearch/indices/analysis/PreBuiltTokenizers.java]: This can also be removed?

**Comment by rjernst** [comment]
Looks good, one more place I noticed, at least in the files changed here.

**Comment by nik9000** [comment]
Fixed that one last place and a few others I caught that were using Elasticsearch index version.

**Comment by nik9000** [comment]
I'll rerun the test suite one last time and merge.

**Pull request merged**